smack-core/src/test/java/org/jivesoftware/smack/packet/XMPPErrorTest.java

```diff
@@ -0,0 +1,40 @@
+/**
+ *
+ * Copyright © 2017 Ingo Bauersachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.packet;
+
+import static org.jivesoftware.smack.packet.XMPPError.Condition;
+import static org.jivesoftware.smack.packet.XMPPError.Type;
+import static org.junit.Assert.assertEquals;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+
+import org.junit.Test;
+
+public class XMPPErrorTest {
+    @Test
+    public void testConditionHasDefaultTypeMapping() throws NoSuchFieldException, IllegalAccessException {
+        Field conditionToTypeField = XMPPError.class.getDeclaredField("CONDITION_TO_TYPE");

```

Comment:
How about making the CONDITION_TO_TYPE field package protected to avoid having to use reflection?